src/design/logger-rate-limiter.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Logger Rate Limiter
+ *
+ * Design a logger system that receive stream of messages along with its timestamps,
+ * each message should be printed if and only if it is not printed in the last 10 seconds.
+ *
+ * Given a message and a timestamp (in seconds granularity),
+ * return true if the message should be printed in the given timestamp, otherwise returns false.
+ *
+ * It is possible that several messages arrive roughly at the same time.
+ *
+ * Example:
+ *
+ * Logger logger = new Logger();
+ *
+ * // logging string "foo" at timestamp 1
+ * logger.shouldPrintMessage(1, "foo"); returns true;
+ *
+ * // logging string "bar" at timestamp 2
+ * logger.shouldPrintMessage(2,"bar"); returns true;
+ *
+ * // logging string "foo" at timestamp 3
+ * logger.shouldPrintMessage(3,"foo"); returns false;
+ *
+ * // logging string "bar" at timestamp 8
+ * logger.shouldPrintMessage(8,"bar"); returns false;
+ *
+ * // logging string "foo" at timestamp 10
+ * logger.shouldPrintMessage(10,"foo"); returns false;
+ *
+ * // logging string "foo" at timestamp 11
+ * logger.shouldPrintMessage(11,"foo"); returns true;
+ */
+
+/**
+ * Initialize your data structure here.
+ */
+class Logger {
+  constructor() {
+    this.map = new Map();
+  }
+  /**
+ * Returns true if the message should be printed in the given timestamp, otherwise returns false.
+        If this method returns false, the message will not be printed.
+        The timestamp is in seconds granularity.
+* @param {number} timestamp
+* @param {string} message
+* @return {boolean}
+*/
+  shouldPrintMessage(timestamp, message) {
+    if (this.map.has(message)) {
```

Comment:
Avoid code repetition by using `&&`:

```javascript
shouldPrintMessage(timestamp, message) {
    if (this.map.has(message) && (timestamp - this.map.get(message) < 10)) {
        // less than 10s elapsed since the message was last sent
        return false;
    } else {
        this.map.set(message, timestamp);
        return true;
    }
}
```

Also, the information on the timestamp's granularity would be better as the parameter's description instead of the function's.